crates/apollo-compiler/src/database/hir.rs

```diff
@@ -22,6 +25,18 @@ pub struct TypeSystemDefinitions {
     pub directives: ByName<DirectiveDefinition>,
 }
 
+/// Contains a `TypeSystemDefinition` together with:
+///
+/// * Other data that can be derived from it.
+/// * Relevant inputs
+#[derive(PartialEq, Eq, Debug)]
+pub struct PrecomputedTypeSystem {
```

Comment:
If we keep this as `Precomputed`, by extension all other slightly computed types (subtype_map) should be prefixed with precomputed, and that's a bit wordy
```suggestion
pub struct TypeSystem {
```

Reply:
Having the struct for "type system definitions, supporting sources, and other stuff derived from definitions" be called just "type system" feels a bit weird, but I don’t really have a better suggestion